### PR TITLE
[RemoteEvent] Ensure consumer is callable

### PR DESCRIPTION
**Pull request by lyrixx**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | yes and no
| New feature?  | no
| Deprecations? | no
| Tickets       |
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/RemoteEvent/Consumer/ConsumerInterface.php
Comment (lyrixx) [inline src/Symfony/Component/RemoteEvent/Consumer/ConsumerInterface.php:1]: was not used at all

File: src/Symfony/Component/RemoteEvent/Messenger/ConsumeRemoteEventHandler.php [hunk 1]
```diff
@@ -33,6 +33,12 @@ public function __invoke(ConsumeRemoteEventMessage $message): void
         }
         $consumer = $this->consumers->get($message->getType());
 
-        $consumer->consume($message->getEvent());
+        if (method_exists($consumer, 'consume')) {
+            $consumer->consume($message->getEvent());
+        } elseif (is_callable($consumer)) {
```
Comment (lyrixx) [inline src/Symfony/Component/RemoteEvent/Messenger/ConsumeRemoteEventHandler.php]: could be nice to use the __invoke method too, like regular messenger handler

File: src/Symfony/Component/RemoteEvent/Messenger/ConsumeRemoteEventHandler.php [hunk 2]
```diff
@@ -33,6 +33,12 @@ public function __invoke(ConsumeRemoteEventMessage $message): void
         }
         $consumer = $this->consumers->get($message->getType());
 
-        $consumer->consume($message->getEvent());
```
Comment (nicolas-grekas) [inline src/Symfony/Component/RemoteEvent/Messenger/ConsumeRemoteEventHandler.php:36]: that's strange, why don't we enforce consumer implement ConsumerInterface instead?

Comment (lyrixx) [reply src/Symfony/Component/RemoteEvent/Messenger/ConsumeRemoteEventHandler.php:36]: I don't know! cc @fabpot

Comment (fabpot) [reply src/Symfony/Component/RemoteEvent/Messenger/ConsumeRemoteEventHandler.php:36]: See #50093

**Pull request closed without merging**